### PR TITLE
Stack support

### PR DESCRIPTION
**Pull request by finnbar**
Introduces Stack support for simply-typed-ext, meaning it can now be run with `stack run`. This automatically runs `alex` and `happy` on the relevant files.

### CODE REVIEW

File: simply-typed-ext.cabal [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+author:         Finnbar Keating and Michael Gale
+maintainer:     f.keating@warwick.ac.uk
+license:        BSD3
+build-type:     Simple
+extra-source-files:
+    README.md
+
+source-repository head
+  type: git
+  location: https://github.com/finnbar/simply-typed-ext
```
Comment (mbg) [inline simply-typed-ext.cabal]: It looks like this file needs to be re-generated from `package.yaml`.

File: package.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+name: simply-typed-ext
+version: 0.1.0.0
+github: "mbg/simply-typed-ext"
+license: BSD3
```
Comment (mbg) [inline package.yaml]: This should be unspecified until I decide on a license for this.
```suggestion
```

File: package.yaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+author: "Michael Gale"
```
Comment (mbg) [inline package.yaml]:
```suggestion
author: "Michael B. Gale"
```

File: package.yaml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+maintainer: "github@michael-gale.co.uk"
+
+extra-source-files:
+- README.md
+
+description: Please see the README on GitHub at <https://github.com/finnbar/simply-typed-ext#readme>
```
Comment (mbg) [inline package.yaml]:
```suggestion
description: Please see the README on GitHub at <https://github.com/mbg/simply-typed-ext#readme>
```

File: package.yaml [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+executables:
+  simply-typed-ext-exe:
+    main: Main.lhs
+    source-dirs: app
+    ghc-options:
+      - -threaded
+      - -rtsopts
+      - -with-rtsopts=-N
+    dependencies:
+      - simply-typed-ext
```
Comment (mbg) [inline package.yaml]: Missing new line at the end of this file.
```suggestion
      - simply-typed-ext

```

File: .gitignore [hunk 6]
```diff
@@ -4,3 +4,5 @@ cabal-dev
 *.hi
 *.chi
 *.chs.h
+
+.stack-work
```
Comment (mbg) [inline .gitignore]: Missing new line:
```suggestion
.stack-work

```

**Review by mbg — CHANGES_REQUESTED** [review]
You have added `.stack-work` to `.gitignore`, but some files from `.stack-work` are part of this PR nonetheless. Could you remove those and make the other changes I have commented on?

**Comment by finnbar** [comment]
Apologies for the silly errors, but those are fixed now.

**Review by mbg — APPROVED** [review]
Thanks, that all looks good to me now.

**Pull request merged**